cached_network_image/lib/src/image_provider/_image_loader.dart

```diff
@@ -4,61 +4,60 @@ import 'dart:ui' as ui;
 import 'dart:ui';
 
 import 'package:cached_network_image_platform_interface/cached_network_image_platform_interface.dart';
-import 'package:flutter/material.dart';
-import 'package:flutter_cache_manager/flutter_cache_manager.dart';
-
 import 'package:cached_network_image_platform_interface'
         '/cached_network_image_platform_interface.dart' as platform
     show ImageLoader;
-import 'package:cached_network_image_platform_interface'
-        '/cached_network_image_platform_interface.dart'
-    show ImageRenderMethodForWeb;
+import 'package:flutter/material.dart';
+import 'package:flutter_cache_manager/flutter_cache_manager.dart';
 
 /// ImageLoader class to load images on IO platforms.
 class ImageLoader implements platform.ImageLoader {
-  @Deprecated('use loadBufferAsync instead')
+  @Deprecated('Use loadImageAsync instead')
   @override
-  Stream<ui.Codec> loadAsync(
+  Stream<ui.Codec> loadBufferAsync(
     String url,
     String? cacheKey,
     StreamController<ImageChunkEvent> chunkEvents,
-    DecoderCallback decode,
+    DecoderBufferCallback decode,
     BaseCacheManager cacheManager,
     int? maxHeight,
     int? maxWidth,
     Map<String, String>? headers,
-    Function()? errorListener,
     ImageRenderMethodForWeb imageRenderMethodForWeb,
-    Function() evictImage,
+    VoidCallback evictImage,
   ) {
     return _load(
       url,
       cacheKey,
       chunkEvents,
-      decode,
+      (bytes) async {
+        final buffer = await ImmutableBuffer.fromUint8List(bytes);
+        return decode(buffer);
+      },
       cacheManager,
       maxHeight,
       maxWidth,
       headers,
-      errorListener,
+      (_) {},
```

Comment:
loadBufferAsync used to have `Function()? errorListener,`, so I think we should make it like this:
```suggestion
      (_) => errorListener(),
```

Reply:
This cannot be done because errorListener has type `void Function(Object)` and not type void Function()

Reply:
But I think it's not very correct. It is better to change the ErrorListener type in loadBufferAsync

Reply:
I see you've changed it already to ` (_) => errorListener?.call(),`, that looks good to me.